### PR TITLE
chore: integrate bundle size tooling

### PR DESCRIPTION
**Pull request by layershifter**
This PR integrates `monosize` bundle size tool (replacement for `@fluentui/bundle-size`).
The benefit is that is also has reporting on PRs now.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
           restore-keys: |
             yarn-cache-folder-
 
-      - run: yarn install --frozen-lockfile
+      - run: yarn install --immutable
```
Comment (layershifter) [inline .github/workflows/ci.yml:38]: `frozen-lockfile` is deprecated

File: monosize.config.js [hunk 2]
```diff
@@ -1,6 +1,12 @@
 const path = require('path');
 
 module.exports = {
+  repository: 'https://github.com/microsoft/griffel',
+  storage: require('monosize-storage-upstash')({
+    url: 'https://us1-nearby-hyena-35747.upstash.io',
+    readonlyToken:
+      'AoujASQgNGQwNGI0YjAtNmExNy00MzNjLWIwNTQtMTIyMGRlODQwZDk2QaCJNVtEfHj6pDLLscOcgMEt-Be1VWcMYbMRR_EqVCw=',
+  }),
```
Comment (layershifter) [inline monosize.config.js:9]: This is an example of adapter for storage. With this config it's possible to compare results with `upstream/main` 🚀

**Comment by layershifter** [comment]
> ## 📊 Bundle size report
> Package & Exports	Baseline (minified/GZIP)	PR	Change
> core
> makeStyles + mergeClasses (runtime)	`19.178 kB`
> `7.27 kB`	`20.016 kB`
> `7.468 kB`	`838 B` 
> `198 B` 
> react
> makeStaticStyles (runtime)	`8.177 kB`
> `3.568 kB`	`8.247 kB`
> `3.595 kB`	`70 B` 
> `27 B` 
> react
> makeStyles + mergeClasses (runtime)	`20.152 kB`
> `7.671 kB`	`21.041 kB`
> `7.89 kB`	`889 B` 
> `219 B` 
> react
> makeStyles + mergeClasses (build time)	`2.731 kB`
> `1.239 kB`	`2.801 kB`
> `1.273 kB`	`70 B` 
> `34 B` 
> Unchanged fixtures
> 🤖 This report was generated against [foo](https://github.com/microsoft/griffel/commit/foo)

This PR does not have regressions, these results are based are based on a report from 2 weeks ago.

File: .github/workflows/bundle-size.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        run: echo ${{ github.event.number }} > pr.txt
+
+      - uses: actions/upload-artifact@v3
+        if: ${{ github.event_name == 'pull_request' }}
+        with:
+          name: monosize-report
+          if-no-files-found: error
+          path: |
+            monosize.md
+            pr.txt
```
Comment (layershifter) [inline .github/workflows/bundle-size.yml:59]: These artifacts are stored for [bundle-size-comment.yml](https://github.com/microsoft/griffel/blob/ecfcdb91f868dd2ec2e1503f832b431001ffa172/.github/workflows/bundle-size-comment.yml)

**Review by ling1726 — APPROVED** [review]

**Pull request merged**